docs/_pages/walk-through.md

```diff
@@ -0,0 +1,555 @@
+
+# Setting up your workspace
+
+In order to get started, make a directory and include in it another directory named 'tests', another directory named 'components', and the following package.json:
+
+{
+  "name": "simulato-lab-tests",
+  "version": "0.1.0",
+  "description": "set of tests for simulato lab",
+  "scripts": {
+    "generate-tests": "sim generate",
+    "run-tests": "sim run -p 5"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "simulato": "0.4.0"
+  }
+}
+
+You will also need a file named config.js with the following:
+
+    'use strict';
+
+    module.exports = {
+        testPath: 'tests',
+        components: 'components',
+        outputPath: 'tests',
+        technique: 'actionFocused',
+    };
+
+Also clone the repo [here](https://www.npmjs.com/package/simulato-test-site) and follow the directions to get the test site up and running as this is what you will be modeling against!
+
+
+# Creating a component
+
+While components are individual pieces of a system, it helps to break down the full system into the components you will need before you start creating any one component.
+
+In the system described above, our actions might be to navigate to the website, click on an article, and close the article. In order to perform these actions, we will need a component to cover navigation, a component to represent the landing page, a component for structures we care about on the landing page (news articles), and a modal representation for the news article:
+
+![](../assets/modal-example.png)
+
+Now that we have our list of components
+* navigation
+* landing page
+* modal
+
+Let's get started with navigation.
+
+## Navigation
+
+A part of running the tests is arriving at our desired location. As we are using selenium to control our browser we will need to first make use of the driver to get there. Fortunately a navigation component is very simple as it doesn't have to deal with the elements or the models of the component and only needs to interact with the actions piece.
+
+Let us go through it piece by piece, first create a file in the components directory 'navigate-to-test-site.js', and include the following:
+
+    'use strict';
+
+    module.exports = {
+        type: 'NavigateToTestSite',
+        entryComponent: {
+            name: 'navigateToTestSite',
+            state: {}
+        },
+        elements(){
+            return [];
+        },
+        model(){
+            return {};
+        },
+        actions(){
+            //below
+        }
+    }
+
+The first thing in our component is the type, the type is the component name, it is how the object of the component will be referenced. Below is the entry component, an entry component is essentially a starting point for the test path. This entry component in particular contains the instanceName of the component and its current state, which is empty, for now think of the state as the current values occupying the model.
```

Comment:
does it make sense to reference the code here again or just point them to the above code snippet by referencing the line number